### PR TITLE
add transformer

### PR DESCRIPTION
**Pull request by xiaowuhu**

### CODE REVIEW

File: Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+解码器层中的各个子模块，如，多头注意力机制，规范化层，前馈全连接都与编码器中的实现相同。
+
+有一个细节需要注意，第一个子层的多头注意力和编码器中完全一致，第二个子层，它的多头注意力模块中，query来自上一个子层，key 和 value 来自编码器的输出。可以这样理解，就是第二层负责，利用解码器已经预测出的信息作为query，去编码器提取的各种特征中，查找相关信息并融合到当前特征中，来完成预测。
+
+#### 模型输出
+
+输出部分就很简单了，每个时间步都过一个 线性层 + softmax层。
+
+线性层的作用：通过对上一步的线性变化得到指定维度的输出，也就是转换维度的作用。转换后的维度对应着输出类别的个数，如果是翻译任务，那就对应的是文字字典的大小。
+
```
Comment (YanjieGao) [inline Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md:264]: Add a 

## 小节与讨论

File: Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
+<!--Copyright © Microsoft Corporation. All rights reserved.
+  适用于[License](https://github.com/microsoft/AI-System/blob/main/LICENSE)版权许可-->
+
+## 2.7 Transformer模型
```
Comment (YanjieGao) [inline Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md:4]: 2.9?

File: Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
```
Comment (YanjieGao) [inline Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md:5]: Please add 2.9 in outline README and link to this file

https://github.com/microsoft/AI-System/blob/main/Textbook/README.md

**Review by YanjieGao — DISMISSED** [review]

File: Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md (hunk 2 above)
Comment (xiaowuhu) [reply Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md:4]: fixed

File: Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md (hunk 1 above)
Comment (xiaowuhu) [reply Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md:264]: fixed

File: Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md (hunk 3 above)
Comment (xiaowuhu) [reply Textbook/第2章-神经网络基础/2.9-注意力机制和Transformer.md:5]: fixed

**Pull request merged**